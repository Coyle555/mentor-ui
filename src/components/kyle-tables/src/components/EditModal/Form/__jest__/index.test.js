jest.mock('../Content', () => {
	return { Content: props => <div>{JSON.stringify(props)}</div> };
});

jest.mock('../Footer', () => {
	return { Footer: props => <div>{JSON.stringify(props)}</div> };
});

import React from 'react';
import { Form } from '../index';
import renderer from 'react-test-renderer';

describe('Rendering edit modal form', () => {
	test('Title of the form', () => {
		const tree = renderer.create(<Form title="Test Record" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Fields in the form', () => {
		const fields = [{ id: 'foo', label: 'Foo' }, { id: 'bar', label: 'Bar' }];
		const tree = renderer.create(<Form fields={fields} />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
