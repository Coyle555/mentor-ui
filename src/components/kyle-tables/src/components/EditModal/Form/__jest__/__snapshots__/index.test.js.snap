// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering edit modal form Fields in the form 1`] = `
<div
  className="edit-form"
  style={
    Object {
      "opacity": 0,
      "transform": "translateX(200px)",
    }
  }
>
  <div
    className="title-container"
  >
    <h2
      className="title"
    >
      
    </h2>
    <i
      className="fal fa-times fa-lg close-form"
    />
  </div>
  <div>
    {"data":{},"fields":[{"id":"foo","label":"Foo"},{"id":"bar","label":"Bar"}]}
  </div>
  <div>
    {"currentIndex":null}
  </div>
</div>
`;

exports[`Rendering edit modal form Title of the form 1`] = `
<div
  className="edit-form"
  style={
    Object {
      "opacity": 0,
      "transform": "translateX(200px)",
    }
  }
>
  <div
    className="title-container"
  >
    <h2
      className="title"
    >
      Test Record
    </h2>
    <i
      className="fal fa-times fa-lg close-form"
    />
  </div>
  <div>
    {"data":{},"fields":[]}
  </div>
  <div>
    {"currentIndex":null}
  </div>
</div>
`;
